### PR TITLE
refactor: Remove WithParams serialization helper, use SER_PARAMS_OPFUNC

### PR DESCRIPTION
**Pull request by maflcko**
Every serialization parameter struct already has the `SER_PARAMS_OPFUNC`, except for one in the tests.

For consistency, and to remove verbose code, convert the test to `SER_PARAMS_OPFUNC`, and use it everywhere, then remove the `WithParams` helper.

### CODE REVIEW

**Comment by fanquake** [comment]
cc @theuni

**Comment by theuni** [comment]
Concept ACK.

**Review by sedited — APPROVED** [review]
ACK fa664dc335f02c392e3268f403f32bcf5bd9eeee

File: src/net_processing.cpp [hunk 1]
```diff
@@ -5374,16 +5374,10 @@ void PeerManagerImpl::MaybeSendAddr(CNode& node, Peer& peer, std::chrono::micros
...
-        ser_enc = CNetAddr::Encoding::V2;
-    } else {
-        msg_type = NetMsgType::ADDR;
-        ser_enc = CNetAddr::Encoding::V1;
-    }
-    m_connman.PushMessage(&node, CNetMsgMaker(node.GetCommonVersion()).Make(msg_type, WithParams(CAddress::SerParams{{ser_enc}, CAddress::Format::Network}, peer.m_addrs_to_send)));
+    const bool wants_addrv2{peer.m_wants_addrv2};
+    const auto& msg_type{wants_addrv2 ? NetMsgType::ADDRV2 : NetMsgType::ADDR};
+    const auto& ser_params{wants_addrv2 ? CAddress::V2_NETWORK : CAddress::V1_NETWORK};
+    m_connman.PushMessage(&node, CNetMsgMaker(node.GetCommonVersion()).Make(msg_type, ser_params(peer.m_addrs_to_send)));
```
Comment (ajtowns) [inline src/net_processing.cpp:5380]: Might have been clearer to write:

```c++
CNetMsgMaker mm(node.GetCommonVersion());
if (peer.m_wants_addrv2) {
    m_connman.PushMessage(&node, mm.Make(NetMsgType::ADDRV2, CAddress::V2_NETWORK(peer.m_addrs_to_send)));
} else {
    m_connman.PushMessage(&node, mm.Make(NetMsgType::ADDR, CAddress::V1_NETWORK(peer.m_addrs_to_send)));
}
```

Was thinking about moving `PushMessage` from connman to cnode though (#28686 etc), so could reconsider this at that time.

**Review by ajtowns — COMMENTED** [review]
ACK fa664dc335f02c392e3268f403f32bcf5bd9eeee

**Comment by ajtowns** [comment]
reACK 99990194ce26af89308fab5ad0b1c8c26e45f80c

**Review by sedited — APPROVED** [review]
Re-ACK 99990194ce26af89308fab5ad0b1c8c26e45f80c

**Pull request merged**